src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/Transaction.java

```diff
@@ -0,0 +1,45 @@
+package uk.nhs.digital.nhsconnect.nhais.model.edifact;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+public class Transaction extends Section {
+    @Getter(lazy = true)
+    private final ReferenceTransactionNumber referenceTransactionNumber =
+        ReferenceTransactionNumber.fromString(extractSegment(ReferenceTransactionNumber.KEY_QUALIFIER));
+    @Getter(lazy = true)
+    private final GpNameAndAddress gpNameAndAddress =
+        GpNameAndAddress.fromString(extractSegment(GpNameAndAddress.KEY_QUALIFIER));
+    @Getter(lazy = true)
+    private final Optional<PersonName> personName =
+        extractOptionalSegment(PersonName.KEY_QUALIFIER).map(PersonName::fromString);
+    @Getter(lazy = true)
+    private final FreeText freeText =
```

Comment:
free text will need to be optional - I'm sure you'll encounter this with the new test data though